processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -685,6 +685,9 @@ IncrementalIndexRowResult toIncrementalIndexRow(InputRow row)
         }
         dimsKeySize += indexer.estimateEncodedKeyComponentSize(dimsKey);
         // Set column capabilities as data is coming in
+        if (dimsKey == null) {
+          capabilities.setIsNullable(NullHandling.sqlCompatible());
```

Comment:
If it's string won't it still count as a null, even in replace-with-default mode?

Btw, consider adding a `NullHandling.isNullable(ValueType)` method if seems useful.

Reply:
This is now handled by the dimension indexers, which now can all determine if they have null values or not, so I don't think `NullHandling.isNullable(ValueType)` is necessary anymore.